### DISCUSSION TITLE
Limited inheritance through sealed

### DISCUSSION THREAD

**Discussion by SamuelEnzi** [dotnet/csharplang#8008] [Language Ideas]
What would you think if it were possible to mark methods and properties as sealed and thus prevent them from being inherited?
Here is a small code example
```csharp
public class A
{
    public int A { get; set; }
    public sealed int As { get; set; }

    public void B() { }
    public sealed void Bs() {}
}

public class B : A
{
    public int VA { get; set; }
}

var B = new B();

B.A // Accessible
B.As // Does not exist
B.B // Accessible
B.Bs // Does not exist
B.VA // Accessible
```

**Comment by HaloFour** [comment]
`sealed` is already used to mark a virtual method as not overridable in derived classes.

What would the purpose be of "hiding" those members?  Couldn't you always upcast back to `A` to invoke them anyway?

**Comment by zms9110750** [comment]
In principle, you cannot prevent them from being inherited. These things must appear in the derived class in order for polymorphism to work properly.
What you want to say is to prevent derived classes from accessing them.
But why?
If you only want to prevent derived classes from using set, then prohibit the permission to set.
But why can external objects be set while their own derived classes cannot?

**Reply by SamuelEnzi** [reply]
Yes, you're right, I hadn't thought of that. I can't remember what I was thinking either.
